src/test/java/org/alfresco/repo/event2/CreateRepoEventIT.java

```diff
@@ -146,4 +147,33 @@ public void testFileNodeType()
         assertTrue("isFile flag should be TRUE for nodeType=cm:content. ", resource.isFile());
         assertFalse("isFolder flag should be FALSE for nodeType=cm:content. ", resource.isFolder());
     }
+
+    @Test
+    public void testCteateMultipleNodesInTheSameTransaction()
+    {
+        retryingTransactionHelper.doInTransaction(() -> {
+            for (int i = 0; i < 3; i++)
+            {
+                nodeService.createNode(
+                    rootNodeRef,
+                    ContentModel.ASSOC_CHILDREN,
+                    QName.createQName(TEST_NAMESPACE, GUID.generate()),
+                    ContentModel.TYPE_CONTENT);
+            }
+            return null;
+        });
+
+        checkNumOfEvents(3);
+
+        RepoEventContainer repoEventsContainer = getRepoEventsContainer();
+        final String eventGroupId1 =
+            getEventData(repoEventsContainer.getEvent(1)).getEventGroupId();
+        final String eventGroupId2 =
+            getEventData(repoEventsContainer.getEvent(2)).getEventGroupId();
+        final String eventGroupId3 =
+            getEventData(repoEventsContainer.getEvent(3)).getEventGroupId();
+
+        //All events in the transaction should have the same eventGroupId
+        assertTrue(eventGroupId1.equals(eventGroupId2) || eventGroupId2.equals(eventGroupId3));
```

Comment:
I forgot to add a comment here earlier, but the logic is wrong, it should be an `&&` condition